builder/builder-source-archive.c

```diff
@@ -580,6 +593,51 @@ builder_source_archive_extract (BuilderSource  *source,
   return TRUE;
 }
 
+static gboolean
+builder_source_archive_bundle (BuilderSource  *source,
+                               BuilderContext *context,
+                               GError        **error)
+{
+  BuilderSourceArchive *self = BUILDER_SOURCE_ARCHIVE (source);
+
+  g_autoptr(GFile) file = NULL;
+  g_autoptr(GFile) download_dir = NULL;
+  g_autoptr(GFile) destination_file = NULL;
+  g_autofree char *download_dir_path = NULL;
+  g_autofree char *file_name = NULL;
+  g_autofree char *destination_file_path = NULL;
+  g_autofree char *app_dir_path = g_file_get_path (builder_context_get_app_dir (context));
+  gboolean is_local;
+
+  file = get_source_file (self, context, &is_local, error);
+  if (file == NULL)
+    return FALSE;
+
+  download_dir_path = g_build_filename (app_dir_path,
+                                        "sources/downloads",
```

Comment:
You can use two separate parameters for `sources` and `downloads` and let `g_build_filename()` insert the appropriate dir separator.